### PR TITLE
chore: add backend extension installer 

### PR DESCRIPTION
**Pull request by meifans**

### CODE REVIEW

File: packages/vscode-extension/src/debug/teamsfxTaskProvider.ts [hunk 1]
```diff
@@ -270,7 +270,7 @@ export class TeamsfxTaskProvider implements vscode.TaskProvider {
     const ngrokChecker = new NgrokChecker(vscodeAdapter, vscodeLogger, vscodeTelemetry);
     const ngrokBinFolder = ngrokChecker.getNgrokBinFolder();
     return this.createTask(
-      TaskDefinition.ngrokStart(workspaceFolder.uri.fsPath, isSkipped, ngrokBinFolder),
+      TaskDefinition.ngrokStart(workspaceFolder.uri.fsPath, isSkipped, [ngrokBinFolder]),
```
Comment (XiaofuHuang) [inline packages/vscode-extension/src/debug/teamsfxTaskProvider.ts:273]: TaskDefinition.ngrokStart is also used in `packages\cli\src\cmds\preview\preview.ts`

File: packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts [hunk 2]
```diff
@@ -41,14 +40,13 @@ export class NgrokChecker implements DepsChecker {
   }
 
   public getDepsInfo(): Promise<DepsInfo> {
-    const details = new Map<string, string>();
-    details.set(binFolderKey, this.getNgrokBinFolder());
     return Promise.resolve({
       name: ngrokName,
       isLinuxSupported: true,
       installVersion: installPackageVersion,
+      binFolders: [this.getNgrokBinFolder()],
```
Comment (kuojianlu) [inline packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts:47]: Curious why ngrok bin folders contain only one path, but func core tool bin folders contain two path?

File: packages/fx-core/src/common/local/taskDefinition.ts [hunk 3]
```diff
@@ -165,16 +165,20 @@ export class TaskDefinition {
   static ngrokStart(
     workspaceFolder: string,
     skipNgrok: boolean,
-    ngrokBinFolder: string
+    ngrokBinFolder: string[] | undefined
```
Comment (kuojianlu) [inline packages/fx-core/src/common/local/taskDefinition.ts:168]:
```suggestion
    ngrokBinFolders: string[] | undefined
```

Comment (kuojianlu) [inline packages/fx-core/src/common/local/taskDefinition.ts:168]: Why add `undefined`?

File: packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts (hunk 2 above)
Comment (meifans) [reply packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts:47]: You can learn more from `funcToolChecker.getPortableFuncBinFolders()`

File: packages/fx-core/src/common/local/taskDefinition.ts (hunk 3 above)
Comment (meifans) [reply packages/fx-core/src/common/local/taskDefinition.ts]: Consistent with `taskDefinition.backend()`

File: packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts:47]: I know func bin folders handle the npm6 issue. My point is that if this issue happens for ngrok?

File: packages/fx-core/src/common/deps-checker/util/extensionInstaller.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const defaultOutputPath = "bin";
+const defaultCsprojPath = "extensions.csproj";
+
+export async function installExtension(
+  backendRoot: string,
+  dotnetCommand: string,
+  logger: DepsLogger,
+  csprojPath: string = defaultCsprojPath,
+  outputPath: string = defaultOutputPath
+): Promise<void> {
```
Comment (XiaofuHuang) [inline packages/fx-core/src/common/deps-checker/util/extensionInstaller.ts:18]: May be can use the return type `Promise<Result<void, DepsCheckerError>>`?

**Review by XiaofuHuang — APPROVED** [review]

File: packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts (hunk 2 above)
Comment (swatDong) [reply packages/fx-core/src/common/deps-checker/internal/ngrokChecker.ts:47]: My understanding is - func is pointed to the npm top-level bin folder which handled by npm, but ngrok is pointed to the internal binary folder which handled by ngrok post-install script (Lucky that node_modules folder does not have compatible issue).

**Review by swatDong — APPROVED** [review]

**Pull request merged**